catalog/dags/es/create_new_es_index/create_new_es_index_types.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+
+from airflow.models.connection import Connection
+from es.recreate_staging_index.recreate_full_staging_index import (
+    DAG_ID as RECREATE_STAGING_INDEX_DAG_ID,
+)
+
+from common.constants import PRODUCTION, STAGING
+from data_refresh.data_refresh_types import DATA_REFRESH_CONFIGS
+from database.staging_database_restore.constants import (
+    DAG_ID as STAGING_DB_RESTORE_DAG_ID,
+)
+
+
+@dataclass
+class CreateNewIndex:
+    """
+    Configuration object for the create_new_es_index DAG.
+
+    Required Constructor Arguments:
+
+    environment:     str representation of the environment in which to create
+                     the new index
+    blocking_dags:   list of dags with which to prevent concurrency; the
+                     generated create_new_es_index dag will fail immediately if
+                     any of these dags are running.
+    reindex_timeout: timedelta expressing maximum amount of time the reindexing
+                     step may take
+    """
+
+    dag_id: str = field(init=False)
+    es_host: str = field(init=False)
+    environment: str
+    blocking_dags: list
+    reindex_timeout: timedelta = timedelta(hours=12)
+
+    def __post_init__(self):
+        self.dag_id = f"create_new_{self.environment}_es_index"
+
+        # Get the appropriate connection information for this environment.
+        conn = Connection.get_connection_from_secrets(
+            f"elasticsearch_http_{self.environment}"
+        )
+        self.es_host = conn.host
+
+
+CREATE_NEW_INDEX_CONFIGS = {
+    STAGING: CreateNewIndex(
+        environment=STAGING,
+        blocking_dags=[RECREATE_STAGING_INDEX_DAG_ID, STAGING_DB_RESTORE_DAG_ID],
```

Comment:
I was thinking about this too, but it might be a can of worms to implement. That being said, I wonder if we could at least add a test similar to the DAG parsing tests, that checks for DAGs with a `wait_for_external_dag` task and asserts that the blocking DAG has a matching `prevent_concurrency_with_dag` task? I think I'll go ahead and create an issue for that :)